lib/TPP/ToBlockLayoutAndBack.cpp

```diff
@@ -83,13 +84,32 @@ static Value handleLayoutNC_NCnc(Location loc, Value input, Value output,
                             builder);
 }
 
+static Value handleLayout_VNNI(Location loc, Value input, Value output,
+                               ArrayRef<OpFoldResult> tiles, OpBuilder &builder,
+                               bool useAlloc = false) {
+  assert(tiles.size() == 1 && "expect 1 block for VNNI");
+  SmallVector<int64_t> innerDimPos = {0};
+  if (!output)
+    return toPackLayoutImpl(loc, input, tiles, innerDimPos, {}, builder,
+                            useAlloc);
+  return toUnPackLayoutImpl(loc, input, output, tiles, innerDimPos, {},
+                            builder);
+}
+
 // Helper function to pack from NC to NCnc.
 static Value toPackLayoutNC_NCnc(Location loc, Value input,
                                  ArrayRef<OpFoldResult> tiles,
                                  OpBuilder &builder, bool useAlloc = false) {
   return handleLayoutNC_NCnc(loc, input, nullptr, tiles, builder, useAlloc);
 }
 
+// Helper function to pack from NC to NCnc.
```

Comment:
Is this comment correct? If this is the case why we need another method?

Reply:
Updated the comment, it uses a different inner_dim_pos as compared to NCnc.